core/src/main/java/hudson/EnvVars.java

```diff
@@ -388,6 +388,14 @@ public void putIfNotNull(String key, String value) {
         if (value!=null)
             put(key,value);
     }
+
+    /**
+     * Add entire map but filter null values out.
+     */
+    public void putAllExceptNull(Map<String, String> map) {
```

Comment:
```suggestion
    public void putAllNonNull(Map<String, String> map) {
```

This might read better? PR looks fine otherwise.

Reply:
Definitely. Thanks for the comment.